x-pack/plugins/ingest_manager/server/collectors/register.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import { CoreSetup } from 'kibana/server';
+import { getIsFleetEnabled } from './config_collectors';
+import { AgentUsage, getAgentUsage } from './agent_collectors';
+import { getInternalSavedObjectsClient } from './helpers';
+import { PackageUsage, getPackageUsage } from './package_collectors';
+import { IngestManagerConfigType } from '..';
+
+interface Usage {
+  fleet_enabled: boolean;
+  agents: AgentUsage;
+  packages: PackageUsage[];
+}
+
+export function registerIngestManagerUsageCollector(
+  core: CoreSetup,
+  config: IngestManagerConfigType,
+  usageCollection: UsageCollectionSetup | undefined
+): void {
+  // usageCollection is an optional dependency, so make sure to return if it is not registered.
+  // if for any reason the saved objects client is not available, also return
+  if (!usageCollection) {
+    return;
+  }
+
+  // create usage collector
+  const ingestManagerCollector = usageCollection.makeUsageCollector<Usage>({
+    type: 'ingest_manager',
+    isReady: () => true,
+    fetch: async () => {
+      const soClient = await getInternalSavedObjectsClient(core);
```

Comment:
Do you think you can use the method we have in appContextService https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/agents/checkin/state_new_actions.ts/#L33-L50 ?

Reply:
Do I already have access to the app context in `setup()`?

Reply:
Unfortunately no